src/gpu/intel/gemm/jit/include/gemmstone/dsl/tensor.hpp

```diff
@@ -58,11 +55,11 @@ class idx_t {
     public:
         name_t() = default;
         explicit name_t(size_t idx) {
-            gpu_assert(idx < 26);
+            gemm_assert(idx < 26);
```

Comment:
I guess it's `gemm_assert` (and not `dsl_assert`) because it's part of the gemmstone `include` headers, correct?

It'd be nice to have one set of assertions in all gemmstone in the future.

Reply:
That's correct. There are a couple of related issues here:
 * `dsl_assert` is a macro, so we cannot rely on name spaces to make the implementation unique
 * `dsl_assert` is relying on a lot of string manipulation which does not seem like it should be exposed via external headers.
 * Some templated dsl objects (`idx_map_t`) need to be available in the header, so we need an assertion mechanism at the external header level.

Reply:
Thanks, I see the issue. I was thinking about 1) having a common project-level assert macro (say `gemmstone_assert` or `gs_assert`) 2) put auxiliary string manipulation functionality under `gemmstone::impl` or `gemmstone::detail`

Assertion macro vs function helps to 1) reliably skip potentially expensive conditions and string message construction in release builds and 2) have streaming support

impl/detail namespace seems to be a general practice when templates in headers are part of the API

I think it's fine as of now, just wanted to share my thoughts on that.